examples/fifth/main.go

```diff
@@ -53,16 +87,57 @@ func main() {
 		os.Exit(1)
 	}
 
-	sqsCmp, err := newSQSComponent()
+	// Initialise SQS
+	sqsAPI := sqs.New(awsSQSSession)
+	sqsQueueArn, err := createSQSQueue(sqsAPI) // Programatically create a queue for the sake of the example
+	if err != nil {
+		log.Fatalf("failed to create sqs queue: %v", err)
+	}
+
+	sqsCmp, err := createSQSComponent(sqsAPI)
 	if err != nil {
 		log.Fatalf("failed to create sqs component: %v", err)
 	}
 
+	// Initialise SNS
+	snsAPI := sns.New(awsSNSSession)
+	topicArn, err := createSNSTopic(snsAPI) // Programatically create a topic for the sake of the example
+	if err != nil {
+		log.Fatalf("failed to create sns topic: %v", err)
+	}
+	snsPub, err := patronsns.NewPublisher(snsAPI)
+	if err != nil {
+		log.Fatalf("failed to create sns publisher: %v", err)
+	}
+
+	// Route SNS to SQS (subscriber)
+	err = routeSNSToSQS(snsAPI, sqsQueueArn, topicArn)
+	if err != nil {
+		log.Fatalf("failed to route sns to sqs: %v", err)
+	}
+
+	// Run the server
 	srv, err := patron.New(name, version, patron.Components(sqsCmp.cmp))
 	if err != nil {
 		log.Fatalf("failed to create service: %v", err)
 	}
 
+	go func() {
```

Comment:
the whole SNS publishing part should go into the fourth example and replace the existing direct SQS publishing.